include/nbl/builtin/hlsl/workgroup/fft.hlsl

```diff
@@ -0,0 +1,111 @@
+#ifndef _NBL_BUILTIN_HLSL_WORKGROUP_FFT_INCLUDED_
+#define _NBL_BUILTIN_HLSL_WORKGROUP_FFT_INCLUDED_
+
+#include <nbl/builtin/hlsl/subgroup/fft.hlsl>
+#include <nbl/builtin/hlsl/bit.hlsl>
+
+namespace nbl 
+{
+namespace hlsl
+{
+namespace workgroup
+{
+
+template<typename Accessor, typename Scalar, uint32_t WorkgroupSize>
+void store(NBL_REF_ARG(Accessor) sharedmemAccessor, uint32_t threadID, NBL_CONST_REF_ARG(complex_t<Scalar>) lo, NBL_CONST_REF_ARG(complex_t<Scalar>) hi) {
+    sharedmemAccessor.set(threadID + 0 * WorkgroupSize, lo.real());
+    sharedmemAccessor.set(threadID + 1 * WorkgroupSize, hi.real());
+    sharedmemAccessor.set(threadID + 2 * WorkgroupSize, lo.imag());  
+    sharedmemAccessor.set(threadID + 3 * WorkgroupSize, hi.imag());    
+}
+
+template<typename Accessor, typename Scalar, uint32_t WorkgroupSize>
+void load(NBL_CONST_REF_ARG(Accessor) sharedmemAccessor, uint32_t threadID, NBL_REF_ARG(complex_t<Scalar>) lo, NBL_REF_ARG(complex_t<Scalar>) hi) {
+    lo.real(sharedmemAccessor.get(threadID + 0 * WorkgroupSize));
+    hi.real(sharedmemAccessor.get(threadID + 1 * WorkgroupSize));
+    lo.imag(sharedmemAccessor.get(threadID + 2 * WorkgroupSize));  
+    hi.imag(sharedmemAccessor.get(threadID + 3 * WorkgroupSize));    
+}
+
+template<typename Accessor, typename Scalar, uint32_t WorkgroupSize, bool inverse>
+void FFT_loop(NBL_REF_ARG(Accessor) sharedmemAccessor, uint32_t step, uint32_t threadID){
+    const uint32_t sub_ix = threadID & (step - 1);
+    const uint32_t lo_x_ix = impl::bitfieldInsert(threadID, 0u ,sub_ix, 1u);
```

Comment:
don't use `impl::bitfieldInsert` this is super hard to read.

Also needs comments, I remember whats supposed to happen, but nobody else will